public/js/search.js

```diff
@@ -0,0 +1,66 @@
+$(function(config){
+  var algolia = algoliasearch(config.applicationId, config.apiKey);
+  var index = algolia.initIndex(config.indexName);
+
+  hideSearch();
+
+  $('.clear-search img, .search-results-container img').on('click', function() {
+    $('.sidebar-navigation--search input').val('');
+    hideSearch();
+  });
+
+  $('.sidebar-navigation--search input').on('input', function() {
+    var query = $(this).val();
+    onQueryChange(query);
```

Comment:
Nitpick but the naming of this function is a little odd. Usually you see this `onSomethingHappened` format when the functions are methods on an object or class. e.g. the `window` object in the browser has properties like `window.onload` and you set them equal to a function.

When it's just a standalone function, not associated with a particular object or class, you would name it something like `handleQueryChange`. Even this is a little vague and I like to avoid it if possible in favor of something more indicative of what's actually happening when the query changes, like `updateSearchResults`.